src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java

```diff
@@ -658,25 +658,19 @@ private SeaLightsPluginInfo createSeaLightsPluginInfo(
         slInfo.setBuildFilesFolders(foldersToSearch);
         slInfo.setBuildFilesPatterns(patternsToSearch);
 
-        if(additionalProps.get(SL_JVM_PARAMS_KEY)!= null){
-            String jvmParams = (String)additionalProps.get(SL_JVM_PARAMS_KEY);
-            Map<String, String> paramsMap = StringUtils.convertKeyValueStringToMap(jvmParams);
-            slInfo.setSelightsJvmParams(paramsMap);
-        }
+        trySetMapParams(slInfo, additionalProps);
 
-        if(additionalProps.get(BUILD_SCANNER_PARAMS)!= null){
-            String scannerParams = (String)additionalProps.get(BUILD_SCANNER_PARAMS);
-            Map<String, String> paramsMap = StringUtils.convertKeyValueStringToMap(scannerParams);
-            slInfo.setBuildScannerParams(paramsMap);
-        }
+        return slInfo;
+    }
 
-        if(additionalProps.get(TEST_LISTENER_JVM_PARAMS)!= null){
-            String listenerParams = (String)additionalProps.get(TEST_LISTENER_JVM_PARAMS);
-            Map<String, String> paramsMap = StringUtils.convertKeyValueStringToMap(listenerParams);
-            slInfo.setBuildScannerParams(paramsMap);
+    private void trySetMapParams(SeaLightsPluginInfo slInfo, Properties additionalProps){
+        for(String key: MAP_PARAMS_KEYS) {
+            if (additionalProps.get(key) != null) {
+                String listenerParams = (String) additionalProps.get(key);
+                Map<String, String> paramsMap = StringUtils.convertKeyValueStringToMap(listenerParams);
+                slInfo.setBuildScannerParams(paramsMap);
```

Comment:
You are always setting the buildScannerParams.
Is that intentional?
Do you have tests for this code?